src/modules/RA/DataLayer/components/tabs/StyleTab/Style/ColorStyleField.js

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+
+import { makeStyles } from '@material-ui/core/styles';
+import { Field, useField } from 'react-final-form';
+import { SelectInput, RadioButtonGroupInput, BooleanInput, useTranslate } from 'react-admin';
+import randomColor from 'randomcolor';
+
+import { fieldTypes } from '../../../../../DataSource';
+
+import Condition from '../../../../../../../components/react-admin/Condition';
+
+import ColorPicker from '../../../../../../../components/react-admin/ColorPicker';
+import GraduateValue from './GraduateValue';
+import CategorizeValue from './CategorizeValue';
+
+import ColorListField from './ColorListField';
+
+import styles from './styles';
+
+const useStyles = makeStyles(styles);
+
+const ColorStyleField = ({ path, fields, getValuesOfProperty }) => {
+  const classes = useStyles();
+  const translate = useTranslate();
+
+  const [defaultSeed] = React.useState(Math.floor((Math.random() * 100000) + 1));
+  const [defaultValue] = React.useState([randomColor()]);
```

Comment:
I'm not sure if this is relevant but you can put a function as argument of the useState to avoid computations every renders https://reactjs.org/docs/hooks-reference.html#lazy-initial-state

```suggestion
  const [defaultSeed] = React.useState(() => Math.floor((Math.random() * 100000) + 1));
  const [defaultValue] = React.useState(() => [randomColor()]);
```

Reply:
Done